pkg/util/validation/formats.go

```diff
@@ -309,29 +309,29 @@ func (FormatLDAPSearchFilterTemplate) CheckFormat(value interface{}) error {
 	return nil
 }
 
-type FormatLDAPOID struct{}
+type FormatLDAPAttribute struct{}
 
-func (FormatLDAPOID) CheckFormat(value interface{}) error {
+func (FormatLDAPAttribute) CheckFormat(value interface{}) error {
 	str, ok := value.(string)
 	if !ok {
 		return nil
 	}
 
-	// numericoid = number 1*( DOT number )
-	// number  = DIGIT / ( LDIGIT 1*DIGIT )
-	// DIGIT   = %x30 / LDIGIT       ; "0"-"9"
-	// LDIGIT  = %x31-39             ; "1"-"9"
-	// https://datatracker.ietf.org/doc/html/rfc4512#section-1.4
 	if len(str) == 0 {
-		return errors.New("expect non-empty OID")
+		return errors.New("expect non-empty attribute")
 	}
 
-	matched, err := regexp.MatchString(`^(?:[0-9]|[1-9][0-9]+)(?:\.(?:[0-9]|[1-9][0-9]+))+$`, str)
+	// An attribute description option is represented by the ABNF:
+	// options = *( SEMI option )
+	// option = 1*keychar
+	// keychar = ALPHA / DIGIT / HYPHEN
+	// According to https://datatracker.ietf.org/doc/html/rfc4512#section-2.5
+	matched, err := regexp.MatchString(`^[a-zA-Z\d-]+$`, str)
```

Comment:
But attribute name starting with a hyphen is uncommon. Let's use production rule of `keystring` here. Update the comment to reflect this implementation choice.